File: util_go/list/list.go
package list

import "fmt"

type ListNode struct {
	Val int
	Next *ListNode
}

func BuildList(s []int) *ListNode {
	preHead := &ListNode{}
	p := preHead
	for _, v := range s {
		p.Next = &ListNode{Val: v}
		p = p.Next
	}
	return preHead.Next
}

func PrintList(l *ListNode) {
	p := l
	for p != nil {
		fmt.Printf("%v, ", p.Val)
		p = p.Next
	}
	fmt.Println()
}

func IsEqualList(l1, l2 *ListNode) bool {
	if l1 == nil && l2 == nil {
		return true
	}
	if l1 == nil || l2 == nil {
		return false
	}
	if l1.Val != l2.Val {
		return false
	}
	return IsEqualList(l1.Next, l2.Next)
}

File: util_py/list.py
# Definition for singly-linked list.

class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


Node = ListNode


def is_equal_list(l1: ListNode, l2: ListNode):
    if not l1 and not l2:
        return True
    if not l1 or not l2:
        return False
    if l1.val != l2.val:
        return False
    return is_equal_list(l1.next, l2.next)


def print_list(lst: ListNode, length=None):
    res = get_built_in_list(lst)
    print(res[:length])


def build_list(arr):
    head = ListNode(0)
    p = head
    for n in arr:
        p.next = ListNode(n)
        p = p.next
    return head.next


def get_built_in_list(head):
    p = head
    res = []
    while p:
        res.append(p.val)
        p = p.next;
    return res

